File: Java/RecyclerView/app/src/main/java/com/ob/recyclerview/AnimalAdapter.java
package com.ob.recyclerview;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.ob.recyclerview.databinding.RowAnimalBinding;
import java.util.ArrayList;

public class AnimalAdapter extends RecyclerView.Adapter<AnimalAdapter.AnimalHolder>
{
    ArrayList<Animal> animals;

    public AnimalAdapter(ArrayList<Animal> animals)
    {
        this.animals = animals;
    }

    @NonNull
    @Override
    public AnimalHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType)
    {
        RowAnimalBinding rowAnimalBinding = RowAnimalBinding.inflate(LayoutInflater.from(parent.getContext()),parent,false);
        return new AnimalHolder(rowAnimalBinding);
    }

    @Override
    public void onBindViewHolder(@NonNull AnimalHolder holder, int position)
    {
        holder.binding.imgAnimal.setImageResource(animals.get(position).getImage());
        holder.binding.txtAnimal.setText(animals.get(position).getName());
    }

    @Override
    public int getItemCount()
    {
        return animals.size();
    }

    static class AnimalHolder extends RecyclerView.ViewHolder
    {
        RowAnimalBinding binding;

        public AnimalHolder(RowAnimalBinding binding)
        {
            super(binding.getRoot());
            this.binding = binding;
        }
    }
}

File: Kotlin/RecyclerView/app/src/main/java/com/ob/recyclerview/AnimalAdapter.kt
package com.ob.recyclerview

import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import com.ob.recyclerview.databinding.RowAnimalBinding

class AnimalAdapter(private val animals: ArrayList<Animal>): RecyclerView.Adapter<AnimalAdapter.AnimalHolder>()
{
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AnimalHolder
    {
        val binding = RowAnimalBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return  AnimalHolder(binding)
    }

    override fun getItemCount(): Int
    {
        return animals.size
    }

    override fun onBindViewHolder(holder: AnimalHolder, position: Int)
    {
        holder.binding.txtAnimal.text = animals[position].name
        holder.binding.imgAnimal.setImageResource(animals[position].image)

        holder.itemView.setOnClickListener {
            Toast.makeText(holder.itemView.context, "OK", Toast.LENGTH_LONG).show()
        }
    }

    class AnimalHolder(val binding: RowAnimalBinding): RecyclerView.ViewHolder(binding.root)
}